api/v1alpha1/kvmcluster_types.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// KVMClusterSpec defines the desired state of KVMCluster
+type KVMClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+
+	// CalicoNetwork defines the network configuration to be used for Calico CNI.
+	// +optional
+	CalicoNetwork CalicoNetwork `json:"calicoNetwork"`
```

Comment:
I'm not sure if this should go here. Other options would be the ControlPlane, Config, or App CRs.

Reply:
Can we somehow install it as a separate app in the cluster? Via clusterctl maybe? Most of the calico configuration goes into the [calico daemonset](https://github.com/giantswarm/k8scloudconfig/blob/24180da406465115c862681884b8955a2189c9eb/files/k8s-resource/calico-all.yaml#L71) which sounds pretty separate from everything. Or we can treat is as infrastructure requirements and leave it in Cluster.

Reply:
I had a good discussion in SIG Operator about this. `cluster-apps-controller` can already pull the Calico network configuration from the `KubeadmConfig` [here](https://github.com/giantswarm/cluster-apps-operator/blob/c1c82344101423244d93e236e56e4fe74673d272/service/controller/resource/clusterconfigmap/desired.go#L52-L74). That doesn't cover the MTU configuration, but I'm not sure if that's used and I would probably lean towards a custom `CalicoCNIConfig` or something if we end up needing that much configurability. I will therefore drop this field for now unless you disagree.

Reply:
MTU is currently not being set on any `KVMConfig` that I could find so I think we're safe on this one